### PR TITLE
DT-9374 Yellow Ribbon reorder of nodes for a11y 

### PR DESCRIPTION
**Pull request by zacharymorel**
## Description
Reordering nodes around the page, adding and exchanging headings

## Original issue(s)
department-of-veterans-affairs/va.gov-team#9374

## Testing done
Unit, integration and e2e tests passing.
Spun up local env and ran voice over.

## Screenshots
Screen Shot 2021-08-17 at 11 12 58 AM

## Acceptance criteria
- [x] Headings are in order for a11y. 

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/yellow-ribbon/components/SearchResultsPage/index.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ export const SearchResultsPage = ({
 
         {/* Helpful Links */}
         <div className="vads-u-display--none medium-screen:vads-u-display--flex vads-u-flex-direction--column">
-          <h3 className="vads-u-margin-top--2">Helpful links</h3>
+          <h2 className="vads-u-margin-top--2">Helpful links</h2>
```
Comment (joshkimux) [inline src/applications/yellow-ribbon/components/SearchResultsPage/index.js:63]: This change from `h3` to `h2` should not be made until after helpful links have been relocated to after the search results 🙇

**Review by joshkimux — COMMENTED** [review]
Thanks for the quick fix!

File: src/applications/yellow-ribbon/components/SearchResult/index.jsx [hunk 2]
```diff
@@ -129,7 +129,7 @@ export const SearchResult = ({ school, onSearchResultClick }) => (
         className="vads-u-font-size--h3 vads-u-font-weight--bold vads-u-font-family--serif vads-u-margin--0"
         data-e2e-id="result-title"
       >
-        {deriveNameLabel(school)}
+        <h3 className="vads-u-margin--0">{deriveNameLabel(school)}</h3>
```
Comment (kelsonic) [inline src/applications/yellow-ribbon/components/SearchResult/index.jsx:132]: Can we put an h3 inside a dd tag without causing an a11y issue? https://html.spec.whatwg.org/multipage/grouping-content.html#the-dd-element (I genuinely don't know)

**Review by Rleahy22 — APPROVED** [review]

File: src/applications/yellow-ribbon/containers/SearchResults/index.jsx [hunk 3]
```diff
@@ -208,6 +208,7 @@ export class SearchResults extends Component {
         <h2
           className="va-introtext va-u-outline--none vads-u-font-size--lg vads-u-margin-top--1p5 vads-u-font-weight--normal"
           data-display-results-header
+          tabIndex="-1"
```
Comment (kelsonic) [inline src/applications/yellow-ribbon/containers/SearchResults/index.jsx:211]: Is the goal to have this focus last?

**Review by kelsonic — APPROVED** [review]
Had a couple of questions but lgtm! 🎉

File: src/applications/yellow-ribbon/components/SearchResultsPage/index.js (hunk 1 above)
Comment (zacharymorel) [reply src/applications/yellow-ribbon/components/SearchResultsPage/index.js:63]: Okay! Got it, thank you.

File: src/applications/yellow-ribbon/components/SearchResult/index.jsx (hunk 2 above)
Comment (zacharymorel) [reply src/applications/yellow-ribbon/components/SearchResult/index.jsx:132]: Neither do I, @joshkimux , do you have anything thoughts?

File: src/applications/yellow-ribbon/containers/SearchResults/index.jsx (hunk 3 above)
Comment (zacharymorel) [reply src/applications/yellow-ribbon/containers/SearchResults/index.jsx:211]: Yeah, the focus is good, right after a search is fired. 
However, the tabindex is to make sure this is invisible in natural tab order of the page.

File: src/applications/yellow-ribbon/components/SearchResult/index.jsx (hunk 2 above)
Comment (zacharymorel) [reply src/applications/yellow-ribbon/components/SearchResult/index.jsx:132]: FYI, with voice over, it navigates to the h3's just fine when navigating through headings.

**Pull request merged**